Linear Regression - Exercises.ipynb

```diff
@@ -116,64 +106,220 @@
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 5,
    "metadata": {},
-   "outputs": [],
-   "source": []
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "r: 0.9856321341091975\n"
+     ]
+    }
+   ],
+   "source": [
+    "r = (np.sum((x-mean_x)*(y-mean_y)))/(np.sqrt((np.sum((x-mean_x)**2)))*np.sqrt((np.sum((y-mean_y)**2))))\n",
+    "print(\"r: \" + str(r))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "$S_{xy}$ = Summed up deviation from the mean of all points (x AND y) \n",
+    "\n",
+    "$s_{xy}$ = Area, in which all points reside in"
+   ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "5. Again compute SSE using the first term in the formula in the middle of slide 13/40. Compute the sum of squares explained by regression SSR which is the last term in the same formula. Then compute the total sum of squares SST and the coefficient of determination or R-squared $R^2$(slide 14/40) as well as the mean square error MSE (slide 19/40). "
+    "*5. Again compute SSE using the first term in the formula in the middle of slide 13/40. Compute the sum of squares explained by regression SSR which is the last term in the same formula. Then compute the total sum of squares SST and the coefficient of determination or R-squared $R^2$(slide 14/40) as well as the mean square error MSE (slide 19/40).*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\\begin{align*}\n",
+    "    SSE = \\sum_{i=1}^{n}(y_{i}-\\hat{y}_{i})^2\n",
+    "\\end{align*}\n",
+    "\\begin{align*}\n",
+    "    SSR = \\sum_{i}^{n}(\\hat{y}_{i}-\\bar{y})^2\n",
+    "\\end{align*}\n",
+    "\\begin{align*}\n",
+    "    SST = SSE+SSR\n",
+    "\\end{align*}\n",
+    "\n",
+    "\n",
+    "\\begin{align*}\n",
+    "    R^2 = \\frac{SSR}{SST}\n",
+    "\\end{align*}"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "metadata": {},
-   "outputs": [],
-   "source": []
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "SSE: 5.868883792048929\n",
+      "SSR: 199.84540192223682\n",
+      "SST: 205.71428571428575\n",
+      "R^2: 0.971470703788651\n"
+     ]
+    }
+   ],
+   "source": [
+    "hat_y = thet0+thet1*x\n",
+    "\n",
+    "SSE=np.sum(((y-hat_y))**2)\n",
+    "print(\"SSE: \" + str(SSE))\n",
+    "\n",
+    "SSR=np.sum((hat_y-mean_y)**2)\n",
+    "print(\"SSR: \" + str(SSR))\n",
+    "\n",
+    "SST = SSE + SSR\n",
+    "print(\"SST: \" + str(SST))\n",
+    "\n",
+    "R_sq = SSR/SST\n",
+    "print(\"R^2: \" + str(R_sq))"
+   ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "6. Give a 90 % confidence interval for $θ_{0}$ and $θ_{1}$ (slide 20/40)."
+    "*6. Give a 90 % confidence interval for $θ_{0}$ and $θ_{1}$ (slide 20/40).*"
```

Comment:
Should be in the form:
conf(theta_0) = [theta_0+/-t_alpha*s_theta_0] and conf(theta_1) = [theta_1+/-t_alpha*s_theta_1]